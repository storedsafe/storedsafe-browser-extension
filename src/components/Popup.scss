@import '../styles/variables.scss';

.popup {
  background-color: $color-primary-0;
}