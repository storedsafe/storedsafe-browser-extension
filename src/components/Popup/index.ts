export { Popup as Main } from './Popup';
