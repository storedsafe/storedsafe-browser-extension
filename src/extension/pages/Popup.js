import React, { useState, useEffect } from 'react';

export default function Popup(props) {
  const [token, setToken] = useState(null);
  const [fields, setFields] = useState(null);

  const onSubmit = (event) => {
    event.preventDefault();
    setToken(event.target.token.value);
  };

  const handleMessage = (request, sender) => {
    setFields(Object.keys(request.fields).map(key => {
      if (/pass/.test(key)) {
        return <li key={key}>{request.fields[key].replace(/./g,'*')}</li>;
      }
      return <li key={key}>{request.fields[key]}</li>;
    }));
  };

  useEffect(() => {
    browser.runtime.onMessage.addListener(handleMessage);
  }, []);

  useEffect(() => {
    browser.storage.local.get('token')
      .then(values => {
        if(values.token !== undefined) {
          setToken(values.token.value);
        }
      });
  }, []);

  useEffect(() => {
    browser.storage.local.set({
      token: {
        value: token,
        lastRefresh: Date.now(),
      },
    });
  }, [token]);

  return (
    <section>
      <p>Token: {token}</p>
      <form onSubmit={onSubmit}>
        <input type="text" name="token" id="token" />
        <input type="submit" />
      </form>
      <ul>
        {fields}
      </ul>
    </section>
  );
}
