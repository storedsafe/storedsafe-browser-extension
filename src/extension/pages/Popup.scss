@use '../base.scss';

.popup {
  box-sizing: border-box;
  overflow: auto;
  width: 400px;
}

