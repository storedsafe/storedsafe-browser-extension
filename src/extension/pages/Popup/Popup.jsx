import React from 'react';
import Header from '../../components/Header';
import Auth from '../../containers/Auth';

export default function Popup() {
  return (
    <div className="popup">
      <Header />
      <Auth />
    </div>
  );
}
