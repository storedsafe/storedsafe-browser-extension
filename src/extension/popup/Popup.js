import React, { useState } from 'react';
import LoginContainer from './Login/LoginContainer';
import MainContainer from './Main/MainContainer';

import browser from 'webextension-polyfill';

export default function Popup(props) {
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  browser.tabs.getCurrent().then(tab => {
    console.log(tab);
  }).catch(error => {
    console.error(error.message);
  });

  let page;

  if (!isAuthenticated)
    page = <LoginContainer onLogIn={() => setIsAuthenticated(true)} />;
  else
    page = <MainContainer onLogOut={() => setIsAuthenticated(false)} />;

  return (
    <main>
      {page}
    </main>
  );
};

