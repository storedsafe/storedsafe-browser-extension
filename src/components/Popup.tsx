import React, { useState } from 'react'
import { SearchBar, OnSearchCallback } from './common/input/SearchBar'
import './Popup.scss'

interface PopupProps {
  searchResults: Results
  onSearch: (needle: string) => void
}

export const Popup: React.FunctionComponent = () => {
  const [needle, setNeedle] = useState<string>('')

  const onSearch: OnSearchCallback = () => {
    console.log(needle)
  }

  return (
    <section className='popup'>
      <SearchBar
        needle={needle}
        onNeedleChange={setNeedle}
        onSearch={onSearch}
        isLoading={false}
      />
    </section>
  )
}
