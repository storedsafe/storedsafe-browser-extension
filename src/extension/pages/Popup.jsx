import React, { useState, useEffect } from 'react';
import AuthForm from '../components/AuthForm';
import FilterableVaultList from '../components/FilterableVaultList';
import SaveLoginForm from '../components/SaveLoginForm';
import logo from '../../assets/logo.png';
import './Popup.scss';

export default function Popup(props) {
  return (
    <section className="popup">
      <header><img src={logo} alt="StoredSafe" /></header>
      <FilterableVaultList />
      <SaveLoginForm />
      <AuthForm />
    </section>
  );
}
