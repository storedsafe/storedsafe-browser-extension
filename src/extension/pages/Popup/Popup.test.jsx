import React from 'react';
import { shallow } from 'enzyme';
import Popup from './Popup';

test('<Popup />', () => {
  const wrapper = shallow(<Popup />);
  expect(wrapper.find('Header').length).toBe(1);
  expect(wrapper.find('Auth').length).toBe(1);
  expect(wrapper).toMatchSnapshot();
});
