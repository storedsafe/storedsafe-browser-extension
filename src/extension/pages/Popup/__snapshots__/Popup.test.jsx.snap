// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> 1`] = `ShallowWrapper {}`;
